src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -0,0 +1,637 @@
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+import { useURLConfigs } from '@/hooks/useURLConfigs';
+
+import { formMixins } from '@/styles/formMixins';
+import { layoutMixins } from '@/styles/layoutMixins';
+
+import { AlertMessage } from '@/components/AlertMessage';
+import { AssetIcon } from '@/components/AssetIcon';
+import { Button } from '@/components/Button';
+import { Checkbox } from '@/components/Checkbox';
+import { Details } from '@/components/Details';
+import { DiffOutput } from '@/components/DiffOutput';
+import { FormInput } from '@/components/FormInput';
+import { FormMaxInputToggleButton } from '@/components/FormMaxInputToggleButton';
+import { Icon, IconName } from '@/components/Icon';
+import { InputType } from '@/components/Input';
+import { Link } from '@/components/Link';
+import { Output, OutputType } from '@/components/Output';
+import { WithDetailsReceipt } from '@/components/WithDetailsReceipt';
+
+import { calculateCanViewAccount, calculateIsAccountViewOnly } from '@/state/accountCalculators';
+import { getSubaccount } from '@/state/accountSelectors';
+import { useAppSelector } from '@/state/appTypes';
+import { getUserVault } from '@/state/vaultSelectors';
+
+import { MustBigNumber } from '@/lib/numbers';
+
+const renderDiffOutput = ({
+  type,
+  value,
+  newValue,
+  showSign,
+  withDiff,
+}: Pick<
+  Parameters<typeof DiffOutput>[0],
+  'type' | 'value' | 'newValue' | 'withDiff' | 'showSign'
+>) => (
+  <DiffOutput
+    type={type}
+    value={value}
+    newValue={newValue}
+    showSign={showSign}
+    withDiff={withDiff}
+  />
+);
+
+type VaultFormError = {
+  type: AlertType;
+  key: string;
+  short?: string;
+  long?: React.ReactNode;
+};
+
+// errors we don't want to show aggressive visual cues about, just disable submit
+const lightErrorKeys = new Set(['disconnected', 'view-only', 'amount-empty']);
+
+const SLIPPAGE_PERCENT_WARN = 0.01;
+const SLIPPAGE_PERCENT_ACK = 0.01;
+
+type VaultDepositWithdrawFormProps = {
+  initialType?: 'deposit' | 'withdraw';
+  onSuccess?: () => void;
+};
+
+export const VaultDepositWithdrawForm = ({
+  initialType,
+  onSuccess,
+}: VaultDepositWithdrawFormProps) => {
+  const stringGetter = useStringGetter();
+  const { vaultsLearnMore } = useURLConfigs();
+  const isAccountViewOnly = useAppSelector(calculateIsAccountViewOnly);
+  const canViewAccount = useAppSelector(calculateCanViewAccount);
+
+  const { userBalance } = useAppSelector(getUserVault) ?? {};
+  const { freeCollateral, marginUsage } = useAppSelector(getSubaccount) ?? {};
+
+  const [selectedType, setSelectedType] = useState<'deposit' | 'withdraw'>(
+    initialType ?? 'deposit'
+  );
+  const [amount, setAmountState] = useState('');
+  const [isSubmitting] = useState(false);
+  const [currentForm, setCurrentForm] = useState<'input' | 'confirm'>('input');
+  const [slippageAck, setSlippageAck] = useState(false);
+
+  const slippagePercent = 0.05;
```

Comment:
[q] why are `SLIPPAGE_PERCENT_WARN` and `SLIPPAGE_PERCENT_ACK` extracted from the component and this one not?

Reply:
This one is the temporary one, not a real constant like the others, just for displaying something nice until abacus.